pkg/controller/datavolume-controller.go

```diff
@@ -291,6 +294,7 @@ func NewDatavolumeController(
 	if err := addDatavolumeControllerWatches(mgr, datavolumeController); err != nil {
 		return nil, err
 	}
+	go reconciler.startSnapshotCsiCrdDetection()
```

Comment:
I think it would be nicer, rather than creating our own goroutine, let the controller runtime manager deal with it.  See here:  https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/manager.go#L61

Basically just pass a Runnable and when the manager is started, it will create/manage the goroutine.

I believe this will clean up all the funniness in the unit test with the dummy goroutine and passing the additional context

Reply:
We take advantage of this in cdi-operator here:  https://github.com/kubevirt/containerized-data-importer/blob/main/pkg/operator/controller/certrotation.go#L80